test/views/DeclarationDetailsViewSpec.scala

```diff
@@ -557,6 +557,30 @@ class DeclarationDetailsViewSpec extends UnitViewSpec with GivenWhenThen with In
       }
     }
 
+    def createSubmissionWithRejectedAmendment(notificationSummaries: Seq[NotificationSummary]) = {
+      val action = Action("id", AmendmentRequest, now, Some(notificationSummaries), Some(uuid), 1)
+      Submission(
+        uuid,
+        "eori",
+        "lrn",
+        Some(mrn),
+        Some("ducr"),
+        notificationSummaries.reverse.headOption.map(_.enhancedStatus),
+        notificationSummaries.headOption.map(_ => now),
+        Seq(action),
+        latestDecId = Some(uuid)
+      )
+    }
+
+    "display 'Fix and resubmit' button and 'Cancel' link when latest notification is rejected Amendment" in {
+      val notificationSummariesAmendmentRejected = List(dmsrejNotification, dmsdocNotification, dmsctlNotification, acceptedNotification)
+      val view = page(createSubmissionWithRejectedAmendment(notificationSummariesAmendmentRejected))(verifiedEmailRequest(), messages)
+      val button = view.getElementsByClass("govuk-button-group")
+      button.get(0).child(0).text() mustBe messages("declaration.details.fix.resubmit.button")
+      button.get(0).child(1).text() mustBe messages("declaration.details.cancel.amendment")
```

Comment:
I would also test in addition that `child(0)` is an actual button and that `child(1)` is just instead a link